src/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -138,13 +138,9 @@ internal static bool CheckIsReserved(char ch, UriComponents component)
         //
         internal static unsafe string EscapeUnescapeIri(char* pInput, int start, int end, UriComponents component)
         {
-            char[] dest = new char[end - start];
+            ValueStringBuilder dest = new ValueStringBuilder(end - start, true);
```

Comment:
As noted, just construct the ValueStringBuilder with end - start.  If you want to set its length and zero init that space, you can do:
```C#
dest.Length = end - start;
dest.AsSpan().Clear();
```

Reply:
Cleaning implemented for initial allocation and resizing.